lib/configure.js

```diff
@@ -272,6 +272,7 @@ function configure (gyp, argv, callback) {
       argv.push('-Dnode_gyp_dir=' + nodeGypDir)
       argv.push('-Dnode_lib_file=' + release.name + '.lib')
       argv.push('-Dmodule_root_dir=' + process.cwd())
+      argv.push('-Dnode_engine=' + (gyp.opts.node_engine || process.jsEngine || 'v8'))
```

Comment:
If you build `node.exe` using `--node-engine=chakracore`, and using that binary if you are building an add-on, `process.jsEngine` will be automatically populated with `chakracore` and we will be building an add-on compatible with `node-chakracore`.  Likewise for `node-v8`.

Only scenario where things can go wrong is someone build `node-chakracore` and use it to build add-on passing `--node-engine=v8` or vice versa. So in that case, your understanding is correct.